uploader/tagged.go

```diff
@@ -83,59 +139,30 @@ LineLoop:
 			continue
 		}
 
-		key := strconv.Itoa(int(reader.Days())) + ":" + unsafeString(name)
+		nameStr := unsafeString(name)
 
+		days := reader.Days()
+		key := strconv.Itoa(int(days)) + ":" + nameStr
 		if u.existsCache.Exists(key) {
 			continue LineLoop
 		}
 
 		if newTagged[key] {
+			// already processed
 			continue LineLoop
 		}
-		n++
-
-		m, err := urlParse(unsafeString(name))
-		if err != nil {
-			continue
-		}
-
-		newTagged[key] = true
-
-		wb.Reset()
-		tagsBuf.Reset()
-		tag1 = tag1[:0]
-
-		t := fmt.Sprintf("__name__=%s", m.Path)
-		tag1 = append(tag1, t)
-		tagsBuf.WriteString(t)
-
-		// don't upload any other tag but __name__
-		// if either main metric (m.Path) or each metric (*) is ignored
-		ignoreAllButName := u.ignoredMetrics[m.Path] || u.ignoredMetrics["*"]
-		tagsWritten := 1
-		for k, v := range m.Query() {
-			t := fmt.Sprintf("%s=%s", k, v[0])
-			tagsBuf.WriteString(t)
-			tagsWritten++
-
-			if !ignoreAllButName {
-				tag1 = append(tag1, t)
-			}
-		}
 
-		for i := 0; i < len(tag1); i++ {
-			wb.WriteUint16(reader.Days())
-			wb.WriteString(tag1[i])
-			wb.WriteBytes(name)
-			wb.WriteUVarint(uint64(tagsWritten))
-			wb.Write(tagsBuf.Bytes())
-			wb.WriteUint32(version)
-		}
+		n++
 
-		_, err = out.Write(wb.Bytes())
-		if err != nil {
+		if err = u.parseName(nameStr, days, tag1, wb, tagsBuf); err != nil {
+			u.logger.Warn("parse",
+				zap.String("metric", nameStr), zap.String("type", "tagged"), zap.String("name", filename), zap.Error(err),
+			)
+			continue LineLoop
+		} else if _, err = out.Write(wb.Bytes()); err != nil {
 			return n, nil, err
 		}
+		newTagged[key] = true
```

Comment:
Since we are here in any case, we can replace it with `map[string]struct{}`, so the memory consumption will be improved slightly.

Reply:
It will required simular changes in index, points and other upload

Reply:
It's indeed out of the scope then